lib/src/spec/common.dart

```diff
@@ -260,37 +260,81 @@ class TokenRequest {
 
 }
 
+/// Params for rest history
+///
+/// https://docs.ably.io/client-lib-development-guide/features/#RSL2b
+class RestHistoryParams {
+  /// start must be equal to or less than end and is unaffected by the request direction
+  ///
+  /// RLS2b1
+  final DateTime start;
 
-abstract class RestHistoryParams {
-  int start;
-  int end;
-  String direction;
-  int limit;
+  /// end must be equal to or greater than start and is unaffected by the request direction
+  ///
+  /// RLS2b1
+  final DateTime end;
+
+  /// Sorting history backwards or forwards
+  ///
+  /// if omitted the direction defaults to the REST API default (backwards)
+  /// RLS2b2
+  final String direction;
+
+  /// Number of items returned in one page
+  ///
+  /// limit supports up to 1,000 items.
+  /// if omitted the direction defaults to the REST API default (100)
+  /// RLS2b3
+  final int limit;
+
+  RestHistoryParams(
+      {DateTime start,
+      DateTime end,
+      String direction = 'backwards',
+      this.limit = 100})
+      : assert(direction == 'backwards' || direction == 'forwards'),
+        direction = direction,
+        start = start ?? DateTime.fromMillisecondsSinceEpoch(0),
+        end = end ?? DateTime.now();
+}
+
+/// https://docs.ably.io/client-lib-development-guide/features/#RTL10
+class RealtimeHistoryParams extends RestHistoryParams {
+  /// Decides whether to retrieve messages from earlier session.
+  ///
+  /// if true, will only retrieve messages prior to the moment that the channel
+  /// was attached or emitted an UPDATE indicating loss of continuity.
+  bool untilAttach;
+
+  RealtimeHistoryParams(
+      {DateTime start,
+      DateTime end,
+      String direction,
+      int limit,
+      this.untilAttach})
+      : super(start: start, end: end, direction: direction, limit: limit);
 }
 
-abstract class RestPresenceParams {
+class RestPresenceParams {
```

Comment:
This class will be enhanced to match the exact requirement while implementing `RestPresence`

Reply:
If classes are used as interfaces it's better to make the fields getters, and if required add matching setters on demand. 
Introducing fields in classes that might be extended is error prone. 

I'd prefer to have such classes "locked down" by default and open up on demand, instead of later trying to comb through all code to figure out if some of it actually needs to be fields, or needs to be writeable.